spec/forms/applicants/basic_details_form_spec.rb

```diff
@@ -65,6 +65,18 @@
       end
     end
 
+    context 'with an invalid email' do
+      before { attributes[:email] = 'not-a-valid-email' }
+      it 'does not persist model' do
+        expect { subject.save }.not_to change { Applicant.count }
+      end
+
+      it 'generate an error' do
+        subject.save
+        expect(subject.errors[:email]).to be_present
```

Comment:
💭 Consider asserting on the error you're actually expecting to avoid false positives (e.g. the model has an error on the email but it's because it's not present where your expectation is that the format is invalid)